### PR TITLE
Rename neo4j-genai references to neo4j-graphrag

### PR DESCRIPTION
**Pull request by willtai**
# Description

Rename neo4j-genai references to neo4j-graphrag. This prepares the repository for the publication of the new package `neo4j-graphrag`. Please merge this PR after publishing to pypi the changes from this PR https://github.com/neo4j/neo4j-genai-python/pull/128/files

## Type of Change
- [x] New feature
- [ ] Bug fix
- [x] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

> **Note**
>
> Please provide an estimated complexity of this PR of either Low, Medium or High
>
>

Complexity:

## How Has This Been Tested?
- [ ] Unit tests
- [ ] E2E tests
- [ ] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [ ] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [ ] CLA (https://neo4j.com/developer/cla/) has been signed
- [ ] CHANGELOG.md updated if appropriate

### CODE REVIEW

File: docs/source/user_guide_rag.rst [hunk 1]
```diff
@@ -11,11 +11,11 @@ and configuring it according to specific requirements.
 Quickstart
 ************
 
-To perform a GraphRAG query using the `neo4j-genai` package, a few components are needed:
+To perform a GraphRAG query using the `neo4j-graphrag` package, a few components are needed:
 
 1. A Neo4j driver: used to query your Neo4j database.
-2. A Retriever: the `neo4j-genai` package provides some implementations (see the :ref:`dedicated section <retriever-configuration>`) and lets you write your own if none of the provided implementations matches your needs (see :ref:`how to write a custom retriever <custom-retriever>`).
-3. An LLM: to generate the answer, we need to call an LLM model. The neo4j-genai package currently only provides implementation for the OpenAI LLMs, but its interface is compatible with LangChain and let developers write their own interface if needed.
```
Comment (stellasia) [inline docs/source/user_guide_rag.rst:18]: Code snippets below are not updated. More generally, this PR is not updating code in doc?

File: src/neo4j_graphrag/__init__.py [hunk 2]
```diff
@@ -12,3 +12,10 @@
 #  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 #  See the License for the specific language governing permissions and
 #  limitations under the License.
+
+import warnings
```
Comment (stellasia) [inline src/neo4j_graphrag/__init__.py:16]: To be removed, now users will be importing from neo4j-graphrag, so all good.

File: docs/source/user_guide_rag.rst (hunk 1 above)
Comment (willtai) [reply docs/source/user_guide_rag.rst:18]: https://github.com/neo4j/neo4j-genai-python/pull/130/commits/17d9e7d12e8dc7df5d3e9846355d04d65406728e

File: src/neo4j_graphrag/__init__.py (hunk 2 above)
Comment (willtai) [reply src/neo4j_graphrag/__init__.py]: https://github.com/neo4j/neo4j-genai-python/pull/130/commits/76696284a904d7dfc6a1d80dd44926f9ea73a31f

**Review by stellasia — APPROVED** [review]
LGTM🤞

**Pull request merged**